presto-router/src/test/java/com/facebook/presto/router/TestScheduler.java

```diff
@@ -88,14 +106,18 @@ public void testWeightedRandomChoiceScheduler()
 
         HashMap<URI, Integer> hitCounter = new HashMap<>();
 
-        for (int i = 0; i < 1000; i++) {
+        for (int i = 0; i < 100_000; i++) {
             URI target = scheduler.getDestination("test").orElse(new URI("invalid"));
             assertTrue(servers.contains(target));
             assertTrue(weights.containsKey(target));
             hitCounter.put(target, hitCounter.getOrDefault(target, 0) + 1);
         }
 
-        assertTrue(hitCounter.get(new URI("192.168.0.3")) > hitCounter.get(new URI("192.168.0.1")));
+        //testing that ratios between weights roughly equate to ratios between actual destinations, may not pass 100% of the time
+        assertTrue((hitCounter.get(servers.get(1)) / hitCounter.get(servers.get(0)) > 8));
+        assertTrue((hitCounter.get(servers.get(1)) / hitCounter.get(servers.get(0)) < 12));
+        assertTrue((hitCounter.get(servers.get(2)) / hitCounter.get(servers.get(1)) > 8));
+        assertTrue((hitCounter.get(servers.get(2)) / hitCounter.get(servers.get(1)) < 12));
```

Comment:
tests should ideally pass 100% of the time. If we need wiggle room here, use `assertEquals` instead with an epsilon value equal to a reasonable amount of error.